apps/web/src/app/tools/components/ToolTryPageClient.tsx

```diff
@@ -639,115 +547,97 @@ export function ToolPageClient({ toolId, slug }: ToolPageClientProps) {
+            context: lastContext,
+          });
         }
-        hideViewToggle={toolId === "claim-evaluator"}
-        generatePrompt={
-          toolId === "claim-evaluator"
-            ? (input: any) => generateClaimEvaluatorPrompt(input)
-            : undefined
+        return resultRenderer(result as ToolResult);
+      }}
+      exampleInputs={examples.map((ex) => ({
+        label: ex.label,
+        value: ex.values,
+      }))}
+      onBeforeSubmit={(input: Record<string, unknown>) => {
+        // Store statement for math tools
+        if (input.statement) {
+          setLastStatement(input.statement as string);
         }
-        onSaveResult={
-          toolId === "claim-evaluator"
-            ? async (result: any, input: any) => {
-                const response = await fetch("/api/claim-evaluations", {
-                  method: "POST",
-                  headers: { "Content-Type": "application/json" },
-                  body: JSON.stringify({
-                    claim: lastClaim,
-                    context: lastContext || undefined,
-                    summaryMean: result.summary?.mean,
-                    rawOutput: result,
-                    explanationLength: input.explanationLength,
-                    temperature: input.temperature,
-                    prompt: input ? generateClaimEvaluatorPrompt(input) : undefined,
-                  }),
-                });
-
-                if (!response.ok) {
-                  throw new Error("Failed to save evaluation");
-                }
-
-                return response.json();
-              }
-            : undefined
+        // Store claim for claim-evaluator
+        if (input.claim) {
+          setLastClaim(input.claim as string);
         }
-        saveButtonText="Save Evaluation"
-        getSavedResultUrl={
-          toolId === "claim-evaluator"
-            ? (id: string) => `/claim-evaluations/${id}`
-            : undefined
+        // Store context for claim-evaluator
+        if (input.context) {
+          setLastContext(input.context as string);
         }
-      />
-    );
-  }
+        return input;
+      }}
+      submitButtonText={
+        toolId.includes("checker") || toolId.includes("validator")
+          ? "Check"
+          : toolId.includes("extractor")
+            ? "Extract"
+            : "Process"
+      }
+      loadingText={
+        toolId.includes("checker") || toolId.includes("validator")
+          ? "Checking..."
+          : toolId.includes("extractor")
+            ? "Extracting..."
+            : "Processing..."
+      }
+      hideViewToggle={toolId === "claim-evaluator"}
+      generatePrompt={
+        toolId === "claim-evaluator"
+          ? (input: Record<string, unknown>) => generateClaimEvaluatorPrompt(input as { claim: string; context?: string; explanationLength?: number; runs?: number })
+          : undefined
+      }
+      onSaveResult={
+        toolId === "claim-evaluator"
```

Comment:
This seems really hacky. Why is there custom functionality for "claim-evaluator" in the generic try page?